src/components/contextual-basemaps/card.tsx

```diff
@@ -0,0 +1,124 @@
+import { useMemo, ReactElement } from 'react';
+
+import Image, { StaticImageData } from 'next/image';
+
+import cn from 'lib/classnames';
+
+import { basemapAtom } from 'store/map';
+import { basemapContextualAtom } from 'store/map-settings';
+
+import { useRecoilState } from 'recoil';
+
+import { INFO } from 'containers/datasets';
+import { BasemapId } from 'containers/datasets/contextual-layers/basemaps';
+
+import { Checkbox, CheckboxIndicator } from 'components/checkbox';
+import Icon from 'components/icon';
+import Info from 'components/widget-controls/info';
+import { WIDGET_CARD_WRAPPER_STYLE } from 'styles/widgets';
+
+import analyticThumb from 'images/thumbs/analytic.png';
+import darkThumb from 'images/thumbs/btn-dark@2x.png';
+import lightThumb from 'images/thumbs/btn-light@2x.png';
+import satelliteThumb from 'images/thumbs/btn-satellite@2x.png';
+import visualThumb from 'images/thumbs/visual.png';
+
+const THUMBS = {
+  light: lightThumb as StaticImageData,
+  dark: darkThumb as StaticImageData,
+  satellite: satelliteThumb as StaticImageData,
+  planet_medres_analytic_monthly: analyticThumb as StaticImageData,
+  planet_medres_visual_monthly: visualThumb as StaticImageData,
+};
+
+import TICK_SVG from 'svgs/ui/tick.svg?sprite';
+
+import { ContextualBasemapsId, MosaicId } from 'types/widget';
+
+import DateSelect from './date-select';
+
+type CardBasemapContextualProps = {
+  id: BasemapId | ContextualBasemapsId;
+  mosaic_id?: MosaicId;
+  type: 'contextual' | 'basemap';
+  name: string;
+  description?: string;
+  thumb?: string;
+  children?: boolean | ReactElement;
```

Comment:
🤔 what use case do you have for `boolean`? Also, don't type `children` by yourself, use `PropsWithChildren`

Reply:
I think it should be a react component so it scales better but up to this point, I don't even need it. I may change it completely